app/controllers/avo/search_controller.rb

```diff
@@ -118,9 +127,36 @@ def apply_search_metadata(models, avo_resource)
       end
     end
 
-    def belongs_to_field
+
+    private
+
+    def fetch_parent
```

Comment:
Maybe `set_parent` is a better name for this method? Because it sets the `@parent` instance var.